influxdb3_server/src/http.rs

```diff
@@ -437,16 +506,55 @@ where
     }
 }
 
+fn validate_db_name(name: &str) -> DbName {
+    // A valid name:
+    // - Starts with a letter or a number
+    // - Is ASCII not UTF-8
+    // - Contains only letters, numbers, underscores or hyphens
```

Comment:
Would be helpful to move this up into a doc comment, even though its not a `pub` function, is still nice for the LSP.